Assets/_BForBoss/_Core/Scripts/Boss/DerekMissileBehaviour.cs

```diff
@@ -1,9 +1,155 @@
+using Perigon.Utility;
 using Perigon.Weapons;
+using Sirenix.OdinInspector;
+using UnityEngine;
+using Logger = Perigon.Utility.Logger;
 
 namespace BForBoss
 {
     public class DerekMissileBehaviour : BulletBehaviour
     {
+        private enum State
+        {
+            TowardsApex,
+            HomingTarget,
+            AutoPilot
+        }
+
+        [Title("Move Towards Apex Direction")] 
+        [SerializeField, InfoBox("Offset of Height to create a fake visual arc")]
+        private float _heightOffsetForInitialLaunch;
+        [SerializeField]
+        private float _towardsApexSpeed = 20;
+        [SerializeField] 
+        private float _towardsApexRotationalSpeed = 10;
+        [SerializeField]
+        private AnimationCurve _towardsApexSpeedCurve;
+        [SerializeField, InfoBox("Missile moves towards the DIRECTION of apex, not to a specific target")] 
+        private float _towardsApexTime = 2;
+        
+        [Title("Homing Towards Target")] 
+        [SerializeField]
+        private float _homingTargetSpeed = 20;
+        [SerializeField] 
+        private float _homingTargetRotationalSpeed = 10;
+        [SerializeField]
+        private AnimationCurve _homingTargetSpeedCurve;
+        [SerializeField, InfoBox("Distance between missile/player before state change to autopilot")] 
+        private float _homingTargetDistanceStateChange = 10;
+        [SerializeField, InfoBox("Max time before homing missile state changes to autopilot")] 
+        private float _homingTargetTimeToLive;
+
+        [Title("AutoDrive Towards Direction")] 
+        [SerializeField]
+        private float _autoPilotSpeed = 30f;
+        [SerializeField] 
+        private float _autoPilotTimeToLive = 2f;
+
+        [Title("Shared Configurations")]
+        [SerializeField] 
+        private float _timeTakenToReachMaxSpeedCurve = 2;
```

Comment:
Nitpick: Could use Min value/range attribute on this so that we don't have things like 0 or negative values

This var is also being used in division as a denominator as well so preventing 0 would be nice